lib/base-compiler.js

```diff
@@ -2021,6 +2085,8 @@ but nothing was dumped. Possible causes are:
             this.compiler.version = version;
             this.compiler.fullVersion = fullVersion;
             this.compiler.supportsCfg = this.isCfgCompiler(version);
+			// all C/C++ compilers support -E
+            this.compiler.supportsPpView = (this.compiler.lang === 'c' || this.compiler.lang === 'c++');
```

Comment:
I think this is a bit of a fragile switch, maybe we could fetch this from the *.properties files?

Reply:
Unfortunately it looks like checks for AST/CFG functionality use similarly fragile checks. Looking at these was the first thing I did when trying to see what the precedence was for checking this sort of support.

https://github.com/compiler-explorer/compiler-explorer/blob/3148a9784d0c9b20f4060117b6da86f151ec11db/lib/base-compiler.js#L1765-L1781

I do think the C/C++ check is quite reasonable given that all compilers handle -E the same way. Maybe it would be better to do it in .properties files but given that similar support checks aren't done through *.properties I might shy away from diving into that for this PR.

Reply:
Not all compilers "really" handle -E the same way; https://godbolt.org/z/z5Kq6TeoM
But nonetheless, I think this is a good enough solution for now.